Allow GHC 8.0

I get errors such as:

```
Text\XML\Pugi\Foreign\Node.hs:119:18: error:
    * Couldn't match type `forall a. Maybe a' with `Maybe (Node_ l m)'
      Expected type: IO (Maybe (Node_ l m))
        Actual type: IO (forall a. Maybe a)
    * In the expression:
        withNode n
        $ \ p
            -> do { q <- f p;
                    if q == nullPtr then
                        return Nothing
                    else
                        Just . Node <$> newForeignPtr finalizerNode q }
      In an equation for `nodeCommon':
          nodeCommon n f
            = withNode n
              $ \ p
                  -> do { q <- f p;
                          .... }
    * Relevant bindings include
        f :: Ptr (n k m) -> IO (Ptr (Node_ l m))
          (bound at Text\XML\Pugi\Foreign\Node.hs:119:14)
        n :: n k m (bound at Text\XML\Pugi\Foreign\Node.hs:119:12)
        nodeCommon :: n k m
                      -> (Ptr (n k m) -> IO (Ptr (Node_ l m))) -> IO (Maybe (Node_ l m))
          (bound at Text\XML\Pugi\Foreign\Node.hs:119:1)
```

--- a/Text/XML/Pugi/Foreign/Node.hs
+++ b/Text/XML/Pugi/Foreign/Node.hs
@@ -1,6 +1,5 @@
 {-# LANGUAGE ForeignFunctionInterface #-}
 {-# LANGUAGE TypeSynonymInstances #-}
-{-# LANGUAGE ImpredicativeTypes #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE KindSignatures #-}

--- a/pugixml.cabal
+++ b/pugixml.cabal
@@ -44,7 +44,7 @@ test-suite test
   build-depends:       base
                      , bytestring         >=0.10 && <0.11
                      , pugixml
-                     , tasty              >=0.10 && <0.11
+                     , tasty              >=0.10 && <0.12
                      , tasty-hunit        >=0.9  && <0.10
   type:                exitcode-stdio-1.0
   ghc-options:         -Wall -O2 -threaded

--- a/pugixml.cabal
+++ b/pugixml.cabal
@@ -28,10 +28,10 @@ library
                        Text.XML.Pugi.Foreign.Attr
   -- other-modules:       
   -- other-extensions:    
-  build-depends:       base               >=4.6  && <4.9
+  build-depends:       base               >=4.6  && <4.10
                      , bytestring         >=0.10 && <0.11
                      , data-default-class >=0.0  && <0.1
-                     , template-haskell   >=2.8  && <2.11
+                     , template-haskell   >=2.8  && <2.12
   -- hs-source-dirs:      
   c-sources:           pugixml-1.4/pugixml.cpp cbit/wrapper.cc
   include-dirs:        pugixml-1.4

